docs/src/developers_guide/contributing_tests.rst

```diff
@@ -0,0 +1,263 @@
+
+How and when to group tests within classes can be based on personal opinion,
+we do not deem consistency on this a vital concern.
+
+`Mocks <https://docs.pytest.org/en/stable/how-to/monkeypatch.html>`_
+--------------------------------------------------------------------
+
+Any mocking should be done with ``pytest.mock``, and monkeypatching where suitable.
+
+.. note::
+    If you think we're missing anything important here, please consider creating an
+    issue or discussion and share your ideas with the team!
+
+Naming Test Classes and Functions
+---------------------------------
+
+Within a test module each tested method may have corresponding test classes,
+for example:
+
+* ``Test_<name of public method>``
+* ``Test_<name of public method>__<aspect of method>``
+
+Within these test classes, the test methods must be named according
+to the aspect of the tested method which they address.
+
+**Examples**:
+
+All unit tests for :py:class:`iris.cube.Cube` reside in:
+
+    :literal:`lib/iris/tests/unit/cube/test_Cube.py`
+
+Within that file the tests might look something like:
+
+.. code-block:: python
+
+    # A single test for the Cube.xml() method.
+    def test_xml_some_general_stuff(self):
+        ...
+
+
+    # A single test for the Cube.xml() method, focussing on the behaviour of
+    # the checksums.
+    def test_xml_checksum_ignores_masked_values(self):
+        ...
+
+
+    # Tests for the Cube.add_dim_coord() method.
+    class Test_add_dim_coord:
+        def test_normal_usage(self):
+            ...
+
+        def test_coord_already_present(self):
+            ...
+
+When testing a function all the tests must reside in the module:
+
+    :literal:`lib/iris/tests/unit/<fully/qualified/module>/test_<function_name>.py`
+
+Within this test module there may be test classes, for example:
+
+* ``Test``
+* ``TestAspectOfFunction``
+
+Within those test classes, the test methods must be named according
+to the aspect of the tested function which they address.
+
+**Examples**:
+
+All unit tests for :py:func:`iris.experimental.raster.export_geotiff`
+must reside in:
+
+    :literal:`lib/iris/tests/unit/experimental/raster/test_export_geotiff.py`
+
+Within that file the tests might look something like:
+
+.. code-block:: python
+
+    # Tests focussing on the handling of different data types.
+    class TestDtypeAndValues:
+        def test_int16(self):
+            ...
+
+        def test_int16_big_endian(self):
+            ...
+
+
+    # Tests focussing on the handling of different projections.
+    def test_no_ellipsoid(self):
+        ...
+
+Unlike unit tests, there is no fixed naming scheme for integration
+tests. But folders and files must be created as required to help
+developers locate relevant tests. It is recommended they are named
+according to the capabilities under test, e.g.
+``metadata/test_pp_preservation.py``, and not named according to the
+module(s) under test.
```

Comment:
More appropriate to be under `Test Categories`, since it is not about how to write a test.

Reply:
You're right that they still deserve a mention here, if only to say "write them how you want".